calculator_test.go

```diff
@@ -2,23 +2,185 @@ package calculator_test
+		{name: "Add two nonunits", fn: calculator.Add, a: 1, b: 2, want: 3},
+		{name: "Add unit and nonunit", fn: calculator.Add, a: 0, b: 2, want: 2},
+		{name: "Add negative and inverse", fn: calculator.Add, a: -2, b: 2, want: 0},
+		{name: "Add negative and positive", fn: calculator.Add, a: -2, b: 3, want: 1},
+		{name: "Add two numbers", fn: calculator.Add, a: 1, b: 2, want: 3},
+		{name: "Add three numbers", fn: calculator.Add, a: 1, b: 2, rest: []float64{3}, want: 6},
+		{name: "Add four numbers", fn: calculator.Add, a: 1, b: 2, rest: []float64{3, 4}, want: 10},
+		{name: "Add five numbers", fn: calculator.Add, a: -1, b: 2, rest: []float64{-3, 4, -5}, want: -3},
+		{name: "Subtract two positives", fn: calculator.Subtract, a: 4, b: 2, want: 2},
+		{name: "Subtract positive and zero", fn: calculator.Subtract, a: 2, b: 0, want: 2},
+		{name: "Subtract zero and positive", fn: calculator.Subtract, a: 0, b: 2, want: -2},
+		{name: "Subtract two negatives", fn: calculator.Subtract, a: -1, b: -2, want: 1},
+		{name: "Subtract two numbers", fn: calculator.Subtract, a: 1, b: 2, want: -1},
+		{name: "Subtract three numbers", fn: calculator.Subtract, a: 1, b: 2, rest: []float64{3}, want: -4},
+		{name: "Subtract four numbers", fn: calculator.Subtract, a: 1, b: 2, rest: []float64{3, 4}, want: -8},
+		{name: "Subtract five numbers", fn: calculator.Subtract, a: -1, b: 2, rest: []float64{-3, 4, -5}, want: 1},
+		{name: "Multiply two positives", fn: calculator.Multiply, a: 2, b: 2, want: 4},
+		{name: "Multiply positive and unit", fn: calculator.Multiply, a: 2, b: 1, want: 2},
+		{name: "Multiply positive and zero", fn: calculator.Multiply, a: 2, b: 0, want: 0},
+		{name: "Multiply zero and zero", fn: calculator.Multiply, a: 0, b: 0, want: 0},
+		{name: "Multiply zero and negative", fn: calculator.Multiply, a: 0, b: -1, want: 0},
+		{name: "Multiply positive and negative units", fn: calculator.Multiply, a: 1, b: -1, want: -1},
+		{name: "Multiply positive and negative unit", fn: calculator.Multiply, a: 2, b: -1, want: -2},
+		{name: "Multiply two negative units", fn: calculator.Multiply, a: -1, b: -1, want: 1},
+		{name: "Multiply negative and negative unit", fn: calculator.Multiply, a: -2, b: -1, want: 2},
+		{name: "Multiply two negatives", fn: calculator.Multiply, a: -2, b: -2, want: 4},
+		{name: "Multiply two numbers", fn: calculator.Multiply, a: 1, b: 2, want: 2},
+		{name: "Multiply three numbers", fn: calculator.Multiply, a: 1, b: 2, rest: []float64{3}, want: 6},
+		{name: "Multiply four numbers", fn: calculator.Multiply, a: 1, b: 2, rest: []float64{3, 4}, want: 24},
+		{name: "Multiply five numbers", fn: calculator.Multiply, a: -1, b: 2, rest: []float64{-3, 4, -5}, want: -120},
+	}
+	for _, tc := range tcs {
+		got := tc.fn(tc.a, tc.b, tc.rest...)
+		if tc.want != got {
+			t.Errorf("%v(%f, %f, %v): want %v, got %v", runtime.FuncForPC(reflect.ValueOf(tc.fn).Pointer()).Name(), tc.a, tc.b, tc.rest, tc.want, got)
```

Comment:
Ah, now I see what problem you're trying to solve with `reflect`. Actually, the name of the test case already contains the function in question, doesn't it?

Reply:
It does, but the test case name could get out of date with the code, or otherwise be wrong.  Won't the function name always be accurate?